### PR TITLE
Inline some private helper methods in ColorbarBase + small refactors.

### PR DESCRIPTION
**Pull request by anntzer**
I think _patch_ax, _config_axis, and _config_axes are short enough that
inlining them improve readability by avoiding having to scroll through
to find the helper's implementation.

Also:
- Disable axes frame_on and navigability in the constructor -- no
  need to do it again and again when drawing.
- Instead of going through the datalim machinery to compute the colorbar
  axes limits, directly compute them from the patch/outline coordinates.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
I understand inlining `_config_axis`, but I'd rather wait until the deprecation period was over so we don't have two copies of the same code lying around.

**Comment by anntzer** [comment]
fair enough, deinlined it.

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -501,17 +499,23 @@ def draw_all(self):
...
-        # decide minor/major axis
-        self._config_axis()
-        self._config_axes(X, Y)
+
+        self._config_axis()  # Inline it after deprecation elapses.
+        # Configure axes limits, patch, and outline.
+        xy = self._outline(X, Y)
+        xmin, ymin = xy.min(axis=0)
+        xmax, ymax = xy.max(axis=0)
+        self.ax.set(xlim=(xmin, xmax), ylim=(ymin, ymax))
```
Comment (QuLogic) [inline lib/matplotlib/colorbar.py:514]: Is this really the same, since it doesn't update `self.ax.dataLim` any more?

Comment (anntzer) [reply lib/matplotlib/colorbar.py:514]: If someone really consults the colorbar axes' datalim then no it's not, but in practice the datalim are only used (should only be used...) for computing autoscaled limits, and we're not going to rely on autoscaling the colorbar axes anyways.

**Pull request merged**